huaweicloud/services/aom/resource_huaweicloud_aom_alarm_rule.go

```diff
@@ -0,0 +1,418 @@
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func ResourceAlarmRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAlarmRuleCreate,
+		ReadContext:   resourceAlarmRuleRead,
+		UpdateContext: resourceAlarmRuleUpdate,
+		DeleteContext: resourceAlarmRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"alarm_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 100),
+					validation.StringMatch(regexp.MustCompile(
+						"^[\u4e00-\u9fa5A-Za-z0-9]([\u4e00-\u9fa5-_A-Za-z0-9]*[\u4e00-\u9fa5A-Za-z0-9])?$"),
+						"The name can only consist of letters, digits, underscores (_),"+
+							" hyphens (-) and chinese characters, and it must start and end with letters,"+
+							" digits or chinese characters."),
+				),
+			},
+			"metric_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"namespace": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dimensions": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"period": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ValidateFunc: validation.IntInSlice([]int{
+					60000, 300000, 900000, 3600000,
+				}),
+			},
+			"unit": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 32),
+			},
+			"comparison_operator": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					">=", ">", "<=", "<", "=",
+				}, false),
+			},
+			"statistic": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"maximum", "minimum", "average", "sum", "sampleCount",
+				}, false),
+			},
+			"threshold": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"evaluation_periods": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(1, 5),
+			},
+			"alarm_description": {
```

Comment:
suggest to rename this parameter to **description**

Reply:
fixed